File: python/course1_toolbox/P002_1_MaxPairwiseProductNaive.py
def read_array():
    n = int(input())
    nums = list(map(int, input().split()))
    if len(nums) != n:
        raise ValueError("Number of elements does not match n")
    return nums

def get_max_pairwise_product(nums):
    if len(nums) < 2:
        raise ValueError("Need at least 2 numbers")
    max_product = float("-inf")
    n = len(nums)
    for i in range(n):
        for j in range(i + 1, n):
            max_product = max(max_product, nums[i] * nums[j])
    return max_product


def main():
    nums = read_array()
    print(get_max_pairwise_product(nums))

if __name__ == "__main__":
    main()

File: java/ucsd-dsa/course1_toolbox/src/main/java/edu/ucsd/dsa/toolbox/module1/P002_1_MaxPairwiseProductNaive.java
package edu.ucsd.dsa.toolbox.module1;

import edu.ucsd.dsa.toolbox.utils.FastScanner;

public class P002_1_MaxPairwiseProductNaive {


    private static long[] readArray() {
        FastScanner fs = new FastScanner(System.in);
        int n = fs.nextInt();
        long[] nums = new long[n];
        for (int i = 0; i < n; i++) {
            nums[i] = fs.nextInt();
        }
        return nums;
    }

    private static long getMaxPairwiseProduct(long[] nums) {
        if (nums.length < 2) throw new IllegalArgumentException("Need at least 2 numbers");
        long max_product = Long.MIN_VALUE;
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                max_product = Math.max(max_product, nums[i] * nums[j]);
            }
        }
        return max_product;
    }

    public static void main(String[] args) {
        long[] nums = readArray();
        System.out.println(getMaxPairwiseProduct(nums));
    }
}
